packages/vite/src/node/plugins/terser.ts

```diff
@@ -100,16 +106,27 @@ export function terserPlugin(config: ResolvedConfig): Plugin {
       worker ||= makeWorker()
 
       const terserPath = loadTerserPath(config.root)
-      const res = await worker.run(terserPath, code, {
-        safari10: true,
-        ...terserOptions,
-        sourceMap: !!outputOptions.sourcemap,
-        module: outputOptions.format.startsWith('es'),
-        toplevel: outputOptions.format === 'cjs',
-      })
-      return {
-        code: res.code!,
-        map: res.map as any,
+      try {
+        const res = await worker.run(terserPath, code, {
+          safari10: true,
+          ...terserOptions,
+          sourceMap: !!outputOptions.sourcemap,
+          module: outputOptions.format.startsWith('es'),
+          toplevel: outputOptions.format === 'cjs',
+        })
+        return {
+          code: res.code!,
+          map: res.map as any,
+        }
+      } catch (e) {
+        e.loc = {
+          file: chunk.fileName,
+          line: e.line,
+          column: e.col,
+        }
+        e.frame =
```

Comment:
```suggestion
        if (e.line !== undefined && e.col !== undefined) {
          e.loc = {
            file: chunk.fileName,
            line: e.line,
            column: e.col,
          }
        }
        if (e.pos !== undefined) {
          e.frame = generateCodeFrame(code, e.pos)
        }
```

Maybe we can do this so the error is enhanced a bit more defensively in case the shape is different.